pkg/bucketcache/bucketcache.go

```diff
@@ -55,3 +85,7 @@ func (b *BucketCache[K, V]) GetBucket(k K) []V {
 	res, _ := b.cache.Get(k)
```

Comment:
We need to copy this slice as well hmm.

Reply:
Good call, updated code to return a copy